client/state/data-layer/wpcom/read/sites/posts/follow/index.js

```diff
@@ -41,50 +41,44 @@ export function requestConversationFollow( { dispatch, getState }, action ) {
 			actionWithRevert
 		)
 	);
-}
+};
 
-export function receiveConversationFollow( store, action, response ) {
+export const receiveConversationFollow = ( action, response ) => {
 	// validate that it worked
 	const isFollowing = !! ( response && response.success );
 	if ( ! isFollowing ) {
-		receiveConversationFollowError( store, action );
-		return;
+		return receiveConversationFollowError( action );
 	}
 
-	store.dispatch(
-		successNotice( translate( 'The conversation has been successfully followed.' ), {
-			duration: 5000,
-		} )
-	);
-}
+	return successNotice( translate( 'The conversation has been successfully followed.' ), {
+		duration: 5000,
+	} );
+};
 
-export function receiveConversationFollowError(
-	{ dispatch },
-	{ payload: { siteId, postId }, meta: { previousState } }
-) {
-	dispatch(
+export function receiveConversationFollowError( {
+	payload: { siteId, postId },
```

Comment:
The concept of `action.payload` is not used much in the rest of Calypso -- we just set properties directly on the action object these days. It comes from [Flux Standard Actions](https://github.com/redux-utilities/flux-standard-action) spec.

Reply:
Let's do it in another PR 👍 it affects more files than related to this PR.